eslint.config.mjs

```diff
@@ -0,0 +1,143 @@
+import { fileURLToPath } from 'node:url';
+
+import eslint from '@eslint/js';
+import eslintPluginImport from 'eslint-plugin-import';
+import eslintPluginJest from 'eslint-plugin-jest';
+import globals from 'globals';
+import tseslint from 'typescript-eslint';
+
+const __filename = fileURLToPath(import.meta.url);
+
+const c = tseslint.config(
+  // Files to ignore
+  {
+    ignores: [
+      '**/dist',
+      'examples',
+      'scratch',
+      'coverage',
+      'packages/core/types/test/*.js',
+      'packages/react-native/android/build/reports',
+      'packages/**/features',
+      'packages/**/fixtures',
+      'test/browser',
+      'test/node',
+      'test/react-native-cli/features/fixtures',
+      'packages/react-native/ios/vendor'
+    ],
+  },
+  eslint.configs.recommended,
+  {
+    rules: {
+      'no-undef': 'warn',
```

Comment:
where lint errors arose I have changed the rule to warn. These are to be reviewed later